spec/requests/api/v2/providers/sync_courses_with_search_and_compare_spec.rb

```diff
@@ -0,0 +1,107 @@
+  let(:user)         { create(:user) }
+  let(:organisation) { create(:organisation, users: [user]) }
+  let(:payload)      { { email: user.email } }
+  let(:token)        { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+  let(:site) { build(:site) }
+  let(:dfe_subject) { build(:subject, subject_name: "primary") }
+  let(:non_dfe_subject) { build(:subject, subject_name: "secondary") }
+  let(:findable_site_status_1) { build(:site_status, :findable, site: site) }
+  let(:findable_site_status_2) { build(:site_status, :findable, site: site) }
+  let(:suspended_site_status) { build(:site_status, :suspended, site: site) }
+  let(:syncable_course) { build(:course, site_statuses: [findable_site_status_1], subjects: [dfe_subject]) }
+  let(:suspended_course) { build(:course, site_statuses: [suspended_site_status], subjects: [dfe_subject]) }
+  let(:invalid_subject_course) { build(:course, site_statuses: [findable_site_status_2], subjects: [non_dfe_subject]) }
+  let(:provider) {
+    create(:provider,
+           organisations: [organisation],
+             courses: [syncable_course, suspended_course, invalid_subject_course],
+             sites: [site])
+  }
+
+  describe 'POST ' do
+    let(:status) { 200 }
+    let(:has_synced) { true }
+
+    before do
+      stub_request(:put, "#{Settings.search_api.base_url}/api/courses/")
+        .to_return(
+          status: status,
+        )
+      allow(SearchAndCompareAPIService::Request).to receive(:sync).with([syncable_course]).and_return(has_synced)
+    end
+
+    subject do
+      post sync_path, headers: { 'HTTP_AUTHORIZATION' => credentials }
+      response
+    end
+
+    let(:sync_path) do
+      "/api/v2/providers/#{provider.provider_code}/sync_courses_with_search_and_compare"
+    end
+
+    context 'when unauthenticated' do
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when user has not accepted terms' do
+      let(:user)         { create(:user, accept_terms_date_utc: nil) }
+      let(:organisation) { create(:organisation, users: [user]) }
+
+      it { should have_http_status(:forbidden) }
+    end
+
+    context 'when unauthorised' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload)           { { email: unauthorised_user.email } }
+
+      it "raises an error" do
+        expect { subject }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when authorized' do
+      context 'current recruitment cycle ' do
+        let(:sync_path) do
+          "/api/v2/recruitment_cycles/#{provider.recruitment_cycle.year}" +
+            "/providers/#{provider.provider_code}/sync_courses_with_search_and_compare"
+        end
+
+        context 'when a successful external call to search has been made' do
+          it { should have_http_status(:ok) }
```

Comment:
This isn't testing that a request is actually made to search-and-compare, either by checking that the WebMock stub was called, or that `has_sync` was called. Given that this is a request spec, the WebMock should probably be tested.

Reply:
We can also fix this as part of 1836 (https://trello.com/c/Fjw7R37k/1836-add-job-to-publish-providers-courses-to-find), but need to make sure we don't forget. Might be worth just doing it here.